homeassistant/helpers/entity.py

```diff
@@ -378,19 +378,19 @@ def async_turn_off(self, **kwargs):
         return self.hass.loop.run_in_executor(
             None, ft.partial(self.turn_off, **kwargs))
 
-    def toggle(self) -> None:
+    def toggle(self, **kwargs) -> None:
```

Comment:
That is not possible in this way. That is a very big change on core. We had already describe how that will be done. So if we want do that, we need to move that function into all components, so that will be possible to validate the params. It is also bad, that you don't know if the `turn_on` or `turn_off` will be called with correct parameters.